http-server/src/main/java/io/micronaut/http/server/util/NettyJsonFilterBodyParser.java

```
@@ -0,0 +1,78 @@
+import io.micronaut.json.JsonMapper;
+import jakarta.inject.Singleton;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+
+/**
+ * {@link JsonFilterBodyParser} implementation which leverages {@link ServerHttpRequest#byteBody()} API and a  {@link JsonMapper}.
+ * @param <T> Body type
+ */
+@Singleton
+@Requires(bean = JsonMapper.class)
+@Requires(missingBeans = JsonFilterBodyParser.class)
+@Experimental
+final class NettyJsonFilterBodyParser<T> implements JsonFilterBodyParser<T> {
+    private static final Logger LOG = LoggerFactory.getLogger(NettyJsonFilterBodyParser.class);
+    private final JsonMapper jsonMapper;
+
+    NettyJsonFilterBodyParser(JsonMapper jsonMapper) {
+        this.jsonMapper = jsonMapper;
+    }
+
+    @Override
+    public Publisher<T> parseBody(HttpRequest<?> request, Class<T> type) {
+        if (!supportsRequestContentType(request)) {
+            return Publishers.empty();
+        }
+        if (request instanceof ServerHttpRequest<?> serverHttpRequest) {
+            return parseBodyInServerHttpRequest(serverHttpRequest, type);
+        }
+        return Publishers.empty();
+    }
+
+    private Mono<T> parseBodyInServerHttpRequest(@NonNull ServerHttpRequest<?> request, Class<T> type) {
+        try (CloseableByteBody closeableByteBody = request.byteBody().split(ByteBody.SplitBackpressureMode.FASTEST)) {
+            return Mono.fromFuture(closeableByteBody.buffer())
+                    .mapNotNull(bb -> {
+                        try {
+                            return jsonMapper.readValue(bb.toByteArray(), type);
+                        } catch (IOException e) {
+                            if (LOG.isErrorEnabled()) {
+                                LOG.error("could not bind JSON body to {}", type);
+                            }
+                            return null;
```

I don't think we should be swallowing JSON parsing exceptions and logging them as errors. If the request is invalid it should be a bad request

---

I changed this to https://github.com/micronaut-projects/micronaut-core/pull/11272/commits/7cdfd5db1058d7c09e928e0423726b490d951e8c

```java
.flatMap(bb -> {
                            try {
                                return Mono.just(jsonMapper.readValue(bb.toByteArray(), Argument.mapOf(String.class, Object.class)));
                            } catch (IOException e) {
                                if (LOG.isErrorEnabled()) {
                                    LOG.error("could not bind JSON body to Map");
                                }
                                return Mono.error(e);
                            }
                        });
```